### ISSUE TITLE
MediaElch is slow - Non-responsive windows

### ISSUE THREAD

**Issue by pokemaster974** [Komet/MediaElch#1718]
**Describe the bug**
Executing action like marked as seen, saving, is followed by non responsive windows and very slowing application.

**To Reproduce**
1. Open MediaElch
2. Do some actions
3. Wait

**Expected behavior**
Not having these non responsive windows.

**Screenshots**
![image](https://github.com/Komet/MediaElch/assets/28442699/1329c9ec-ece1-422e-b39e-31e426b197ad)
![image](https://github.com/Komet/MediaElch/assets/28442699/9b09d7c7-2f42-44e4-aaef-70339805aaf7)

**MediaElch Version:**
 - [ ] 2.10.6 (stable)

**Operating System:**
 - [ ] Windows 10

**Additional context**
In the issue #878 there's something about an issue with temporary cache image. In my case, I saw that my folder have already 5000 images but even after closing the app images don't seem to be deleted. In the other issue, there's about 90k images, but in my case after clicking on about 300 movies (to mark them as seen) I've already a lot of images in the local AppData folder. Is it expected ?

To be honest, I don't know why but sometimes there's no slowing windows, and something action like saving can take about 10 minutes, without changing nothing in my processus.

**Comment by pokemaster974** [comment]
> Sadly a known issue. File operations are _synchronous_/_blocking_ as of now. If you have a slow filesystem or use a network drive, MediaElch will slow down a lot. I've put effort in trying to make more coding asynchronous (see e.g. [f2122c6](https://github.com/Komet/MediaElch/commit/f2122c69f1c8fedbe40157ae851a425d0138be6d) ).
> 
> Thanks for reporting.

Thanks for the reply. If you feel that my issue needs to be closed, feel free to close it.
Regards.

**Comment by bugwelle** [comment]
No, you're right about the performance issue, of course! I will keep it open. :)

**Comment by hermannx5** [comment]
Just wanted to confirm the issue. Strangely it's sometimes way worse. After restarting Mediaelch, performance is better and at least no freezes. But still, unfortunately it is an issue also for me...

**Comment by greenfedora** [comment]
A particular occurrence of this issue: 
1. Put a new episode in a TV show's directory on a network drive
2. Right click on the show and select "Search for new episodes"
3. Show updates quickly
4. Open the show and season
5. Attempt to select the new episode

Mediaelch becomes unresponsive.  System monitor show that it is reading something big over the network, 10MB/s for several minutes. 

Why does it need to load anything at this point?  Is there a work around?
